src/vision/mod.rs

```diff
@@ -78,8 +79,13 @@ macro_rules! feat_func_def {
             unsafe {
                 let mut temp: i64 = 0;
                 let err_val = $ffi_name(&mut temp as MutAfArray, self.feat as Feat);
+
+                let temp_array = Array::from(temp);
+                let retained = temp_array.clone();
+                unsafe { mem::forget(temp_array); }
```

Comment:
I just changed the implementation based on your suggestion.
Using `mem::forget` here feels like a bit of a hack, but doing it this way doesn't require touching other files.